packages/discordx/src/types/public/common.ts

```diff
@@ -76,6 +81,13 @@ export type EventOptions = {
   botIds?: string[];
 };
 
+/**
+ * Slash choice type
+ */
+export type SlashChoiceType = {
```

Comment:
This should be written as follows.

```ts
export type SlashChoiceType = Record<string, string | number>;
```

Reply:
It _could_ be written like that as well. This is opinionated, and it is somewhat explained why in [this SO post](https://stackoverflow.com/questions/54100025/difference-between-index-signature-and-record-for-empty-object).

One main benefit of using index signature (possibly the only benefit) is that you can name the index like below: `key`.

Point is, either is good, there's not one you _should_ use over the other - it's up to you.

Reply:
accepted